gdnative-core/src/core_types/quat.rs

```diff
@@ -9,22 +12,248 @@ pub struct Quat {
+    #[inline]
+    pub fn get_euler(self) -> Vector3 {
+        let basis = Mat3::from_quat(self.glam());
+        let elm = basis.to_cols_array_2d();
+        // Copied from Basis::get_euler_yxz
+        let m12 = elm[1][2];
+        if m12 < 1.0 - CMP_EPSILON as f32 {
+            if m12 > CMP_EPSILON as f32 - 1.0 {
+                #[allow(clippy::float_cmp)] // Godot also used direct comparison
+                if elm[1][0] == 0.0
+                    && elm[0][1] == 0.0
+                    && elm[0][2] == 0.0
+                    && elm[2][0] == 0.0
+                    && elm[0][0] == 1.0
+                {
+                    Vector3::new((-m12).atan2(elm[1][1]), 0.0, 0.0)
+                } else {
+                    Vector3::new(
+                        (-m12).asin(),
+                        elm[0][2].atan2(elm[2][2]),
+                        elm[1][0].atan2(elm[1][1]),
+                    )
+                }
+            } else {
+                Vector3::new(FRAC_PI_2, elm[0][1].atan2(elm[0][0]), 0.0)
+            }
+        } else {
+            Vector3::new(-FRAC_PI_2, (-elm[0][1]).atan2(elm[0][0]), 0.0)
+        }
+    }
+
+    /// Returns the inverse of the quaternion.
+    #[inline]
+    pub fn inverse(self) -> Self {
+        Self::gd(self.glam().inverse())
+    }
+
+    /// Returns `true` if this quaternion and `quat` are approximately equal, by running
+    /// `is_equal_approx` on each component
     #[inline]
-    pub fn is_equal_approx(self, to: &Self) -> bool {
+    pub fn is_equal_approx(self, to: Self) -> bool {
         self.x.is_equal_approx(to.x)
             && self.y.is_equal_approx(to.y)
             && self.z.is_equal_approx(to.z)
             && self.w.is_equal_approx(to.w)
     }
 
+    /// Returns whether the quaternion is normalized or not.
+    #[inline]
+    pub fn is_normalized(self) -> bool {
+        self.glam().is_normalized()
+    }
+
+    /// Returns the length of the quaternion.
+    #[inline]
+    pub fn length(self) -> f32 {
+        self.glam().length()
+    }
+
+    /// Returns the length of the quaternion, squared.
+    #[inline]
+    pub fn length_squared(self) -> f32 {
+        self.glam().length_squared()
+    }
+
+    /// Returns a copy of the quaternion, normalized to unit length.
+    #[inline]
+    pub fn normalized(self) -> Self {
+        Self::gd(self.glam().normalize())
+    }
+
+    /// Sets the quaternion to a rotation which rotates around axis by the specified angle, in
+    /// radians. The axis must be a normalized vector.
+    #[inline]
+    pub fn set_axis_angle(&mut self, axis: Vector3, angle: f32) {
+        *self = Self::from_axis_angle(axis, angle)
+    }
+
+    /// Sets the quaternion to a rotation specified by Euler angles (in the YXZ convention: when
+    /// decomposing, first Z, then X, and Y last), given in the vector format as (X angle, Y angle,
+    /// Z angle).
+    #[inline]
+    pub fn set_euler(&mut self, euler: Vector3) {
```

Comment:
The above two methods are rather useless in my opinion -- it's literally just `a = Quat::from_axis_angle(...)` instead of `a.set_axis_angle(...)`. Godot might provide them, because in GDScript you cannot choose between value and reference semantics.

It's also inconsistent with the whole rest of the API which returns new copies, and as such could lead to confusion.

I would remove both.